docs: update readme.md for more clarity (#11)
canonical/kubeflow-roles-operator: README.md

### Kubeflow User Aggregation Roles

> Kubeflow uses the concept of [user facing roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles) to provide access control to users.  Kubeflow's instantiation of this pattern is described [here](https://github.com/kubeflow/manifests/tree/3e08dc102059def5a0b0d04560c7d119959bf506/common/kubeflow-roles) upstream.  The `ClusterRoles` Kubeflow implements are are:

Reviewer: ```suggestion
Kubeflow uses the concept of [user facing roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles) to provide access control to users.  Kubeflow's instantiation of this pattern is described [here](https://github.com/kubeflow/manifests/tree/3e08dc102059def5a0b0d04560c7d119959bf506/common/kubeflow-roles).  The `ClusterRoles` Kubeflow implements are are:
```

* rbac.authorization.kubeflow.org/aggregate-to-kubeflow-view
* rbac.authorization.kubeflow.org/aggregate-to-kubeflow-edit
* rbac.authorization.kubeflow.org/aggregate-to-kubeflow-admin

These `ClusterRoles` ([see here](https://github.com/canonical/kubeflow-roles-operator/blob/d96c15e4de8bb36e9ec039ae66c12af1084ecd2b/src/manifests/kubeflow-roles.yaml#L4) for their definition in this charm) aggregate the permissions of all existing `ClusterRoles` that match a selector (in this case, `ClusterRoles` that have one of the `rbac.authorization.kubeflow.org/aggregate-to-kubeflow-*: "true"` labels), allowing permissions to be added to the aggregation roles without editing the roles themselves.  

> Permissions aggregated into these `ClusterRoles` cover both basic Kubernetes resources, such as defined in [this ClusterRole](https://github.com/canonical/kubeflow-roles-operator/blob/d96c15e4de8bb36e9ec039ae66c12af1084ecd2b/src/manifests/kubeflow-roles.yaml#L74), as well as any access that a custom application requires.  Applications in Kubeflow often add new `CustomResources` to the Kubernetes cluster that users must interact with, such as the `Notebook` object from the `notebook-controller`.  To grant user access to these custom resources, applications must [implement their own `ClusterRoles`](https://github.com/canonical/kubeflow-roles-operator/blob/afe3e1ea0a6dcb4136a506d4d2b697f9d1589a27/src/manifests/notebook-controller.yaml#L17) that, through the `label: rbac.authorization.kubeflow.org/aggregate-to-kubeflow-*` labels, are aggregate into the above described roles.  

Reviewer: ```suggestion
Permissions aggregated into these `ClusterRoles` cover both standard Kubernetes resources (e.g. `Pods` and `Services`), such as defined in [this ClusterRole](https://github.com/canonical/kubeflow-roles-operator/blob/d96c15e4de8bb36e9ec039ae66c12af1084ecd2b/src/manifests/kubeflow-roles.yaml#L74), as well as any access to custom resources.  Applications in Kubeflow often add new `CustomResources` to the Kubernetes cluster that users must interact with, such as the `Notebook` object from the `notebook-controller`.  To grant user access to these custom resources, applications must [implement their own `ClusterRoles`](https://github.com/canonical/kubeflow-roles-operator/blob/afe3e1ea0a6dcb4136a506d4d2b697f9d1589a27/src/manifests/notebook-controller.yaml#L17) that, through the `label: rbac.authorization.kubeflow.org/aggregate-to-kubeflow-*` labels, are aggregate into the above described roles.  
```
